src/model.hpp

```diff
@@ -220,16 +213,18 @@ class model {
 
     // only thread safe if called outside the run() method
     void add_artificial_spike(cell_member_type source, time_type tspike) {
-        current_spikes().get().push_back({source, tspike});
+        if (domain_.is_local_gid(source.gid)) {
```

Comment:
I think there is no need at this point — it does what it says on the tin. It's likely this will go away when we have artificial spike sources in the model, but maybe it'd be useful to keep for tests. Let's not anticipate that yet :)